velox/docs/functions/spark/string.rst

```diff
@@ -427,3 +427,46 @@ String Functions
     Returns string with all characters changed to uppercase. ::
 
         SELECT upper('SparkSql'); -- SPARKSQL
+
+.. spark:function:: varchar_type_write_side_check(input, limit) -> varchar
+   :noindex:
+
+    An implementation for Spark's varcharTypeWriteSideCheck function in CharVarcharCodegenUtils.
+    Ensures length check for ``input`` so that it conforms to Spark's char/varchar semantics.
+    If the number of characters in ``input`` is less than or equal to ``limit``, ``input`` is returned as is.
+    If the number of characters in ``input`` exceeds ``limit``, trailing spaces(0x20) are trimmed to fit the limit.
+    Throws exception if the trimmed string still exceeds ``limit``. ::
+
+        -- Test with SparkSQL that triggers the function
+        create table srcvarchar(id string) stored as parquet;
+        create table tgt(id varchar(3)) stored as parquet;
+        insert into tgt select id from srcvarchar;
+
+.. spark:function:: char_type_write_side_check(input, limit) -> varchar
+   :noindex:
+
+    An implementation for Spark's charTypeWriteSideCheck function in CharVarcharCodegenUtils.
+    Ensures length check for ``input`` so that it conforms to Spark's char/varchar semantics.
```

Comment:
Returns string of length ``limit`` by right padding spaces(0x20) or trailing 0x20(space) characters, throws if not fit in.